docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -242,9 +242,369 @@ This guide will help you create a new Expectation Suite by profiling your data w
+              domains:
+                - my_datetime_domain_1: # QUESTION: Should we allow for multiple domains in a rule? QUESTION: Can / should domains be defined outside of the scope of a specific rule?
+                  # columns of type ``datetime`` (using the ``semantic_types`` field)
+                - my_datetime_domain_2
+                  # columns with the suffix ``_dt``
+              parameters:
+                - $my_dateformat_parameter_1:
+                  # string format
+              expectation_configuration_builders:
+                - expectation: expect_column_values_to_match_strftime_format
+                column: $domains.column # columns for all domains
+                strftime_format: $my_dateformat_parameter_1.parameter.date_string
+
+
+        TODO: insert more examples with actual code.
+
+        TODO: This is an example from James' spec document copied over verbatim, and should be edited. It has been edited as of 20210423
+
+        # QUESTION - what is the $something.something syntax?
+        # PROPOSED ANSWER: $instance_name(parameter_name,domain_name - maybe `domain` is ok and assume the rule domain).attribute(instance or class).sub_attribute.sub_sub_attribute...
+        # TODO: DomainBuilders are 1:1, not Domains
+
+
+        # NOTE: PLEASE SEE BELOW FOR UPDATED CODE AS OF 20210423 ------------------------------------------------------
+
+        .. code-block:: yaml
+
+          variables:
+            false_positive_threshold: 0.01
+          rules:
+            my_rule_for_datetimes:  # "just-a-name"
+              # JPC: what is happening here -- we're asking, "Which columns in this data are datetimes?"
+              domain_builder:
+                name: my_domain_builder_name # QUESTION: This is added - is that OK?
+                class_name: SimpleSemanticTypeColumnDomainBuilder
+                semantic_types:  # AJB: semantic_types are defined in the Domain Builder and referenced here by name
+                                 #      domains are checked against these types for inclusion in the rule domain.
+                                 #      All domains corresponding to any of the semantic_types are included.
+                                 # semantic_type key is determined by the domain builder not global e.g. SimpleSuffixColumnDomainBuilder where suffix is the keyword.
+                  - datetime
+                user_input_list_of_domain_names: # TBD - this is maybe batch_id,domain_id together - probably Typed Object DomainId - can add human readable IDs. Probably too much complexity for first version.
+              parameter_builders:
+                - name: my_dateformat  # This is shorthand for `parameter_builder_name`
+                  # JPC: what is happening here -- we're asking, "What date format matches the data in this column?"
+                  class_name: SimpleDateFormatStringParameterBuilder
+                  module_name: # OPTIONAL
```

Comment:
Could you elaborate on this? What is this meant to represent?

Reply:
`module_name` represents the module where you can find the `SimpleDateFormatStringParameterBuilder`